client/driver/docker.go

```diff
@@ -118,6 +118,22 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 	logger.Printf("[DEBUG] driver.docker: using %d bytes memory for %s", hostConfig.Memory, task.Config["image"])
 	logger.Printf("[DEBUG] driver.docker: using %d cpu shares for %s", hostConfig.CPUShares, task.Config["image"])
 
+	mode := task.Config["network_mode"]
+        // Ignore the container mode for now
+	validModes := map[string]bool{
+		"default": true,
+		"bridge":  true,
+		"none":    true,
+		"host":    true,
+	}
+	if !validModes[mode] {
+		logger.Printf("[WARN] invalid setting for network mode %s, defaulting to bridge mode on docker0", mode)
+	}
+	if mode == "host" && len(task.Resources.Networks) != 0 {
+		panic("task called with network mode = host and extra network configuration, which are not compatible")
```

Comment:
I would definitely not panic here. We can either hoist the validation logic out to a function which returns an error, and call it before `createContainer`, or just not check it and document what that behavior is in that case. If Docker complains when these options collide, that might be an easy way to punt on this.